logback-core/src/main/java/ch/qos/logback/core/recovery/CountingOutputStream.java

```diff
@@ -0,0 +1,48 @@
+package ch.qos.logback.core.recovery;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+class CountingOutputStream extends OutputStream {
+
+  private final OutputStream delegate;
+
+  private long count;
```

Comment:
This should probably be volatile. While there is external synchronization using various locks when calling `write()`, the new state of the variable may not be reflected immediately in other threads after this thread updates it. The increment operations `+=` and `++` are actually each 2 operations, a read and then a write, this means the `count` may be incorrectly updated by another thread afterwards, as the value it reads could be stale.